step/src/main/java/com/google/sps/DatabaseHandler.java

```diff
@@ -69,7 +78,9 @@ public static void addNotification(String firstId, String secondId,
     entity.setProperty("otherUserId", secondId);
     entity.setProperty("timestamp", timestamp);
     entity.setProperty("type", type);
-    datastore.put(entity); 
+    if (firstId != null && secondId != null) {
+      datastore.put(entity);
+    } 
```

Comment:
I think something like if (firstId == null || secondId == null) { // error handling (maybe throw NullException?) } at the top of the method is more efficient so we don't have to run code we're not even going to use. For documentation purposes, can we also add a comment for the method that specifies that the id strings shouldn't be null?